vcpkg/produce/publishing-to-a-git-registry.md

```diff
@@ -0,0 +1,144 @@
+```
+
+## 2 - Create the registry layout
+
+A vcpkg Git-based registry requires the following:
+
+* a folder named `ports` to contain the registry's ports,
+* a folder named `versions` to contain the versions database,
+* a file named `baseline.json`  in the `versions` folder.
+
+Your repository can contain any other files you want, for example: scripts, a LICENSE file, a `README.md`
+file, documentation, etc.
+
+1 - Create the `ports` and `versions` folders:
+
+```Console
+mkdir ports
+mkdir versions
+```
+
+2 - Create a file named `baseline.json` in the `versions` folder with the following contents:
+
+```json
+{
+  "default": {}
+}
+```
+
+The minimal valid `baseline.json` file should include the above contents. Read the [versioning
+documentation](../users/versioning.md) to learn more about the `baseline.json` file and the versions
+database.
+
+3 - Commit your changes to the repository
+
+```Console
+git add .
+git commit -m "Initial commit"
+```
+
+Optionally, if you have a remote Git repository set up, this is a good point to push your changes.
+
+```Console
+git remote add origin https://example.com/vcpkg-registry.git
+git push --set-upstream origin HEAD
+```
+
+## 3 - Add ports to the registry
+
+Copy all your package ports in the `ports` folder in the registry. Each port must have its own
+directory containing at least a `vcpkg.json` file and a `portfile.cmake` file. Learn more about
+creating ports in the tutorial on [packaging GitHub
+repositories](../examples/packaging-github-repos.md).
+
+```Console
+git add ports/.
+git commit -m "Add registry ports"
+```
+
+## 4 - Update the versions database
+
+Each port in the registry must have a corresponding version file. Version files are generated by
+the `vcpkg x-add-version` command. This command also updates the baseline entry for each port in
+`version/baseline.json`.
+
+To update the versions database for all your ports at once, run:
+
+```Console
+vcpkg --x-builtin-ports-root=./ports --x-builtin-registry-versions-dir=./versions x-add-version --all --verbose
+```
+
+The `--x-builtin-ports-root` and `--x-builtin-registry-versions-dir` are redirection options.
+Normally, the `x-add-version` command operates on vcpkg's built-in registry; but by using the
+redirection options, it is possible to use the command on local Git registries.
+
+The `--all` option makes vcpkg scan the `ports` directory for all available ports and updates them
+all at once. Lastly, the `--verbose` option makes the command print each operation it executes to
+standard output, you can suppress the output by removing this option.
+
+The `x-add-version` requires that all port changes have been committed to the repository's Git
+registry. Read the [`x-add-version` command] documentation to learn more.
+
+When all the output looks correct, run:
+
+```Console
+git add .
+git commit -m "Update versions database"
+```
+
+If you have a remote Git repository, don't forget to push your changes:
+
+```Console
+git push
+```
+
```

Comment:
It would probably be a good idea to show how the user can use their registry to verify that they followed the tutorial steps correctly.

Also should we talk about `x-ci-verify-versions`?

Reply:
I planned to have that in a next step action, a tutorial for how to consume private registries.